src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,176 @@
+        for (Map.Entry<K, V> entry : map.entrySet()) {
+            put(entry.getKey(), entry.getValue());
+        }
+    }
+
+    public boolean containsKey(K key) {
+        Node<K, V> node = table[getIndex(key, table.length)];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return true;
+            }
+            node = node.next;
+        }
+        return false;
+    }
+
+    public boolean containsValue(V value) {
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                if (Objects.equals(value, node.value)) {
+                    return true;
+                }
+                node = node.next;
+            }
+        }
+        return false;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = table[getIndex(key, table.length)];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
+        return null;
+    }
+
+    /*
+    wasn't tested
+     */
+    public V remove(K key) {
+        Node<K, V> node = table[getIndex(key, table.length)];
+        if (node == null) {
+            return null;
+        }
+        if (Objects.equals(key, node.key)) {
+            table[getIndex(key, table.length)] = node.next;
+            size--;
+            return node.value;
+        }
+        Node<K, V> next = node.next;
+        while (next != null) {
+            if (Objects.equals(next.key, key)) {
+                node.next = next.next;
+                size--;
+                return next.value;
+            }
+            node = node.next;
+            next = next.next;
+        }
         return null;
     }
 
+    public boolean remove(K key, V value) {
+        if (remove(key) == null) {
+            return false;
+        }
+        return true;
+    }
+
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    public List<V> getValues() {
+        List<V> list = new ArrayList<>();
+        if (size == 0) {
+            return list;
+        }
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                list.add(node.value);
+                node = node.next;
+            }
+        }
+        return list;
+    }
+
+    private void resize() {
+        int newCapacity = table.length * 2;
+        threshold = (int) (threshold * LOAD_FACTOR);
+        Node<K, V>[] newTable = (Node<K, V>[]) new Node[newCapacity];
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                putToTable(node.key, node.value, newTable);
+                node = node.next;
+                size--;
+            }
+        }
+        table = newTable;
+    }
+
+    private void putToTable(K key, V value, Node<K, V>[] table) {
```

Comment:
```suggestion
    private void putToTable(K key, V value) {
```
we can access table from the class, no need to pass it as parameter, if you need 2 tables in resize for example. just create a temp one inside of resize and use it there

Reply:
Thanks, done)